src/Paprika.Tests/MerkleTests.cs

```diff
@@ -0,0 +1,54 @@
+using NUnit.Framework;
+using Paprika.Crypto;
+using Paprika.Store;
+
+namespace Paprika.Tests;
+
+public class MerkleTests
+{
+    private const int SmallDb = 256 * Page.PageSize;
+
+    [Test]
+    public void Empty_database()
+    {
+        using var db = PagedDb.NativeMemoryDb(SmallDb);
+
+        using (var batch = db.BeginNextBatch())
+        {
+            batch.Commit(CommitOptions.FlushDataAndRoot);
+        }
+
+        using (var batch = db.BeginReadOnlyBatch())
+        {
+            var rootHash = batch.GetRootHash();
```

Comment:
I think we should leave the db agnostic in regards to the calculations. As implemented right now in #88 , there is a clear split between blocks that are `latest` or `safe` and blocks that are `finalized`. Only the latter will be stored in the database in a form that makes it squashed together. If you build up the tests maybe it would be useful to build it on level, where you imperatively state `Set for leaves values'. After all, when the left-fold is applied, all the information will be squashed together.

Reply:
I'm not sure I quite follow. You mean separating `computing/setting the hashes` from `get the root hash`? Could you rephrase the last two sentences?

Reply:
I think we could have the batch a bit more agnostic about the `RootHash`. Yes, in #88 there's the metadata, but maybe we could leave it and just operate on the tree with setting `Set(Key.KeccakOrRlp(NibblePath.Empty), ...` ? We can leave as is for now and discuss further. It does not break the idea that much. Let's pause this discussion till #88 lands.